autoload/fzf_mru/actions.vim

```diff
@@ -33,5 +41,5 @@ function! fzf_mru#actions#mru(...) abort
 
   let extra = extend(copy(get(g:, 'fzf_layout', {'down': '~40%'})), options)
 
-  call fzf#run(fzf#wrap('name', extra, 0))
+  call fzf#run(fzf#wrap('name', extend(extra, s:p()), 0))
```

Comment:
Im trying to wrap my head around that one, but it seams to be broken.

Let's assume extra is `{'options': ['--layout=reverse', '--info=inline']}`, now running `extend({'options': ['--layout=reverse', '--info=inline']}, {'options': ['--preview']})` will result with `{'options': ['--preview']}` where `{'options': ['--layout=reverse', '--info=inline', '--preview']}` should be expected. This is intent of this function, to make sure it will preserve original options.

When you run `fzf#vim#with_preview({'options': ['--layout=reverse', '--info=inline']})` it will result with correct output.

Reply:
For this code and the code in the next comment I have to admit that I don't remember.  I got the code from somewhere. I am certain I understood it at the time.  If nothing else, I'll see if I can find out where I got the code from.

Reply:
Based on the commit comment I'm guessing I got this code from the fzf.vim plugin.

Reply:
Wherever it is from, its still not working as it should:

```vim
let extra = {'options': ['--layout=reverse', '--info=inline']}
let preview = {'options': ['--preview']}
let expected_result = {'options': ['--layout=reverse', '--info=inline', '--preview']}
" expected result:
echo expected_result
" actual result:
echo extend(extra, preview)
```
Output for this is:
```vim
{'options': ['--layout=reverse', '--info=inline', '--preview']}
{'options': ['--preview']}
```

Which is wrong because you are not using `fzf#vim#with_preview()` as intended.

Correct solution should be something like this here:
```vim
fzf#run(fzf#wrap('name', s:preview(extra), 0)
```
And in `s:preview` function something like that:
```vim
function s:preview(extra)
  " ... get preview_window
  if empty(preview_window)
    return a:extra
  endif
  " when preview_window = ['right:50%', 'ctrl-/']
  let arguments = reverse(add(reverse(g:fzf_preview_window), extra))
  return call('fzf#vim#with_preview', arguments)
endfunction
```
The issue is that extra should be prepended to the `preview_window` variable, as `fzf#vim#with_preview` takes dict as a first argument and then each array element as separate arguments, hence `reverse(add(reverse`. Not sure if there is a better way to prepend array or apply argumetns to the function. This one would work.

Reply:
Unfortunately the value of preview_window is not guaranteed to be a list so your solution is not quite right.  e.g. my value for fzf_preview_window is 'right:60%'.

Reply:
I think the old solution does the prepending does it not?  It seems to when I test it.  The code is 
`  let arguments = add(copy(a:000), preview_window)
a:000 is a list (the list of all arguments passed to s:preview(), if no arguments are passed it is an empty list). A copy is made so that a:000 is modified and then preview_window is appended to it using add().

Reply:
If I write 
```vim
[{'options': ['--layout=reverse', '--info=inline']}, 'right:60%']
```
and use my value for fzf_preview_window everything looks fine AFAICT.
Here is the output:
```vim
[{'options': ['--layout=reverse', '--info=inline']}, 'right:60%']
```
That seems right to me.

Reply:
I'm going to only comment on #17 from now on. It's getting very confusing switching back and forth.